src/features/scroll_to_bottom.js

```diff
@@ -22,13 +22,23 @@ const styleElement = buildStyle(`
 }
 `);
 
+let timeoutID;
+
+const onLoadersAdded = loaders => {
+  if (active) {
+    clearTimeout(timeoutID);
+  }
+};
+
 const scrollToBottom = () => {
+  clearTimeout(timeoutID);
   window.scrollTo({ top: document.documentElement.scrollHeight });
-  const loaders = [...document.querySelectorAll(knightRiderLoaderSelector)];
 
-  if (loaders.length === 0) {
-    stopScrolling();
-  }
+  timeoutID = setTimeout(() => {
+    if (!document.querySelector(knightRiderLoaderSelector)) {
+      stopScrolling();
+    }
+  }, 150);
```

Comment:
I think we can bump this up to be safer; if working with resource-constrained CI has taught me anything, it's that the UI can take way longer to do things it should be doing near-instantly.

500ms feels like a good balance of safe, but still quick in terms of human perception.

```suggestion
  }, 500);
```

Reply:
Yeah—if I recall correctly, in this case, the relevant time seemed to be measured from the start (air quote that) of the synchronous code execution, and so extremely low values worked fine. But if working with Safari has taught me anything, it's that conclusions about the micro aspects of how javascript code works in practice are always a little fragile :D